access/src/main/java/org/apache/accumulo/access/AccessEvaluator.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.access;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * <p>
+ * An implementation of the Accumulo visibility standard as specified in this document (TODO write
+ * the document based on current Accumulo implementation and post somewhere).
+ *
+ * <p>
+ * Below is an example that should print false and then print true.
+ *
+ * <pre>
+ * {@code
+ * var evaluator = VisibilityArbiter.builder().authorizations("ALPHA", "OMEGA").build();
+ *
+ * System.out.println(evaluator.canAccess("ALPHA&BETA"));
+ * System.out.println(evaluator.canAccess("(ALPHA|BETA)&(OMEGA|EPSILON)"));
+ * }
+ * </pre>
+ *
+ *
+ * @since ???
+ */
+public interface AccessEvaluator {
+  /**
+   * @return true if the expression is visible using the authorizations supplied at creation, false
+   *         otherwise
+   * @throws IllegalArgumentException when the expression is not valid
+   */
+  boolean canAccess(String accessExpression) throws IllegalAccessExpressionException;
```

Comment:
Curious if the method overrides should have a default implementation that way the actual implementation only has to implement one method.

```suggestion
  default boolean canAccess(String accessExpression) throws IllegalAccessExpressionException {
    return canAccess(accessExpression.getBytes(UTF_8));
  }
```

Reply:
I would not add that to start with because once a public API method has a default impl in an interface that it can not be removed.  Where if it starts w/o it its ok to add it later.